### DISCUSSION TITLE
How do I include a SCSS file in my Nuxt module?

### DISCUSSION THREAD

**Discussion by dan-hammond** [nuxt/nuxt#20993] [Questions]
How do I include a SCSS file in my Nuxt module please? 

When developing locally, the below works as intended. However, when I package and install the module in a site, it complains that it can't find `node_modules/foo/dist/runtime/main.scss`. 

It looks like I need to either change the resolver to point at `src/runtime/main.scss`, or `dist/runtime/main.css` when packaged, but I don't know how to achieve that. 

```javascript
import { fileURLToPath } from "url";
import {
  defineNuxtModule,
  createResolver
} from "@nuxt/kit";

export default defineNuxtModule( {
  meta: {
    name: "foo",
    version: "1.0",
    configKey: "foo"
  },
  setup ( _, nuxt ) {
    const { resolve } = createResolver( import.meta.url );

    const runtimeDir = fileURLToPath( new URL( "./runtime", import.meta.url ) );
    nuxt.options.css.unshift(
      resolve( runtimeDir, "./main.scss" )
    );
  }
} );
```

Thanks

**Comment by maou-shonen** [comment]
https://github.com/nuxt/nuxt/discussions/20546
Does this answer not apply to you?

**Reply by dan-hammond** [reply]
I'm not sure. It's definitely related to my issue, but I'm not sure it solves it. I have already edited my package.json to include the src styles, and can see they're preserved when installed in node_modules. 

However, I don't know how to reliably reference those src styles in a Nuxt module, as the resolver in my example above will point to the dist folder when it's built, not the src folder containing my SCSS files.

When running locally it seems to want the path to be something like:
`resolve( runtimeDir, "./main.scss" )`

but when running as an installed package, it instead wants:
`resolve( runtimeDir, "../src/main.scss" )`
OR
`resolve( runtimeDir, "./main.css" )`

I would prefer to be referencing it as CSS in the consuming websites as then they don't need Sass installed.

**Reply by edgarsn** [reply]
#20551 problem nr. 2 in my issue. So far no response on it, but I know nuxt authors are busy. This is blocker for me to finish module.

**Comment by dan-hammond** [comment]
A solution we've come up with is the following:

```javascript
let mainStyleFile = "./main.css";

// Bit of a hack to ensure we use the right style file in dev and when packaged
if ( runtimeDir.endsWith( "src/runtime" ) || runtimeDir.endsWith( "src\\runtime" ) ) {
  mainStyleFile = "./main.scss";
}

nuxt.options.css.unshift(
  resolve( runtimeDir, mainStyleFile ),
);
```

I'm not sure if this is the ideal solution, but it seems to work for us.

**Reply by edgarsn** [reply]
I confirm. This workaround also works for me.

**Reply by davide-granello** [reply]
where and how is `runtimeDir` defined?

**Reply by dan-hammond** [reply]
I haven't tested the below, but I think either one of these approaches should work:

```typescript
import { createResolver } from "@nuxt/kit";
// ...
const { resolve } = createResolver( import.meta.url );
const runtimeDir = resolve( "./runtime" );
```

```typescript
import { fileURLToPath } from "url";
// ...
const runtimeDir = fileURLToPath( new URL( "./runtime", import.meta.url ) );
```
I think the first one is probably the preferred Nuxt way of doing it.

**Reply by davide-granello** [reply]
thank you! In my case, i made it work with:
```ts
setup(options, nuxt) {
    const resolver = createResolver(import.meta.url)
    const runtimeDir = resolver.resolve('./runtime')

    const isDevelopment = runtimeDir.endsWith('src/runtime') || runtimeDir.endsWith('src\\runtime')

    const extension = isDevelopment ? 'scss' : 'css'

    const treeviewStyle = './Components/Treeview/style.' + extension
    nuxt.options.css.unshift(resolver.resolve(runtimeDir, treeviewStyle))

    // ....

  }
```

**Comment by RomanSkrypnik** [comment]
I'm having the same issue and It has become quite a problem
I use `tailwindcss` in my `nuxt module` and I'm getting my scss files converted into css ones as well (In playground everything works fine)

It causes problems while building project because `tailwind directives` such as: `@apply`, `@layer,` `@tailwind` are not processed by `tailwind compiler`:

![image](https://github.com/nuxt/nuxt/assets/44501848/ca069fae-5220-43c7-a142-bd41f8398565)
I have also noticed `nuxt/ui` module solution: https://github.com/nuxt/ui/blob/07a4d13c0fcb05c87fb42e02a3a2d6c5c52ccf09/src/runtime/ui.config/layout/card.ts

@danielroe, is there any other way to solve this problem?
